pennylane/templates/embeddings.py

```diff
@@ -295,6 +295,228 @@ def BasisEmbedding(features, wires):
+          "got {}.".format(n_wires, len(features))
+    _check_shape(features, (n_wires,), bound='max', msg=msg)
+
+    msg = "Option for local field not known. Has to be one of ``'X'``, ``'Y'``, or ``'Z'``."
+    _check_hyperp_is_in_options(local_field, ['X', 'Y', 'Z'], msg=msg)
+    if local_field == 'Z':
+        local_field = RZ
+    elif local_field == 'X':
+        local_field = RX
+    else:
+        local_field = RY
+
+    repeat = _check_number_of_layers([weights])
+
+    weights = np.array(weights)
+    weights_shape = weights.shape
+    if n_wires == 1:
+        msg = "QAOAEmbedding with 1 qubit and {} layers requires weight " \
+              "array of shape {}; got {}".format(repeat, (repeat, 1), weights_shape)
+        _check_shape(weights, (repeat, 1), msg=msg)
+    elif n_wires == 2:
+        msg = "QAOAEmbedding with 2 qubits and {} layers requires weight " \
+              "array of shape {}; got {}".format(repeat, (repeat, 3), weights_shape)
+        _check_shape(weights, (repeat, 3), msg=msg)
+    else:
+        msg = "QAOAEmbedding with {} qubits and {} layers requires weight " \
+              "array of shape {}; got {}".format(n_wires, repeat, (repeat, 2*n_wires), weights_shape)
+        _check_shape(weights, (repeat, 2*n_wires), msg=msg)
+    #####################
+
+    for l in range(repeat):
+
+        # encode inputs into RX gates
+        for i in range(n_wires):
+            # Either feed in feature
+            if i < n_features:
+                RX(features[i], wires=wires[i])
+            # or a Hadamard
+            else:
+                Hadamard(wires=wires[i])
+
+        # trainable "Ising" ansatz
+        if n_wires == 1:
+            local_field(weights[l, 0], wires=wires[0])
+        elif n_wires == 2:
+            CNOT(wires=[wires[0], wires[1]])
+            RZ(2 * weights[l, 0], wires=wires[0])
+            CNOT(wires=[wires[0], wires[1]])
+
+            # local fields
+            for i in range(n_wires):
+                local_field(weights[l, i+1], wires=wires[i])
+        else:
+            for i in range(n_wires):
+                if i < n_wires - 1:
+                    CNOT(wires=[wires[i], wires[i + 1]])
+                    RZ(2 * weights[l, i], wires=wires[i])
+                    CNOT(wires=[wires[i], wires[i + 1]])
+                else:
+                    # enforce periodic boundary condition
+                    CNOT(wires=[wires[i], wires[0]])
+                    RZ(2 * weights[l, i], wires=wires[i])
+                    CNOT(wires=[wires[i], wires[0]])
+            # local fields
+            for i in range(n_wires):
+                local_field(weights[l, n_wires + i], wires=wires[i])
+
+    # repeat feature encoding once more at the end
+    for i in range(n_wires):
+        # Either feed in feature
+        if i < n_features:
+            RX(features[i], wires=wires[i])
+        # or a Hadamard
+        else:
+            Hadamard(wires=wires[i])
```

Comment:
This would be a call then to the ``encode_inputs_into_rx_gates()`` function (or whichever name is picked if refactored).